x-pack/plugins/fleet/server/services/security/uninstall_token_service/index.ts

```diff
@@ -189,11 +198,15 @@ export class UninstallTokenService implements UninstallTokenServiceInterface {
   }
 
   public async getToken(id: string): Promise<UninstallToken | null> {
-    const namespacePrefix = this.soClient.getCurrentNamespace()
-      ? `${this.soClient.getCurrentNamespace()}:`
-      : '';
-    const filter = `${UNINSTALL_TOKENS_SAVED_OBJECT_TYPE}.id: "${namespacePrefix}${UNINSTALL_TOKENS_SAVED_OBJECT_TYPE}:${id}"`;
-    const tokenObjects = await this.getDecryptedTokenObjects({ filter });
+    const useSpaceAwareness = this.isScoped && (await isSpaceAwarenessEnabled());
+    const namespaceFilter = useSpaceAwareness
+      ? getNamespaceFiltering(this.soClient.getCurrentNamespace() ?? DEFAULT_SPACE_ID)
```

Comment:
On line 676, the SO client is being initialized without the spaces extension. The return value for the call to `getCurrentNamespace` with no parameter will always be undefined. What is the intended behavior here?

Reply:
We use that non space aware so client only when our service is not scoped (`this.isScoped = false`) (for background task, fleet setup) and if the service is not scoped we are not doing filtering see the line above. This is covered by integration test, but I could add some unit tests to make that behaviour more obvious.

Reply:
Ahh...thank you! I missed that.